.circleci/config.yml

```diff
@@ -0,0 +1,54 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/golang:latest
+    steps:
+      - run:
+          name: Prepare golang testing tools
+          command: |
+            go get -u github.com/jstemmer/go-junit-report
+            go get -u github.com/golangci/golangci-lint/cmd/golangci-lint
+      - checkout
+      - run:
+          name: install
+          command: go install ./...
+      - run:
+          name: test and take coverage
+          command: |
+            mkdir -p .circleci/test/gotest
+            go test -v -race -coverprofile=coverage.out ./... | go-junit-report > .circleci/test/gotest/results.xml;
+      # - run:
+      #     name: upload coverage
+      #     command: bash <(curl -s https://codecov.io/bash)
+      - run:
+          name: search lints
+          command: golangci-lint run
+
+  release:
+    docker:
+      - image: circleci/golang:latest
+    steps:
+      - checkout
+      - run:
+          name: run goreleaser
+          command: curl -sL https://git.io/goreleaser | bash
+
+workflows:
+  version: 2
+  commit:
+    jobs:
+      - build:
+          filters:
+            branches:
+              only: /.*/
+            tags:
+              only: /.*/
+      - release:
+          requires:
+            - build
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v[0-9]+(\.[0-9]+){2}$/
```

Comment:
If we want to release ichigo with goreleaser, set a tag v#.#.# and push it.
i.e.

```
git checkout master
git tag v1.0.2
git push --tags
```